File: c++/leetcode/222. Count Complete Tree Nodes.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution
{
public:
  int count(TreeNode *, int);
  int countNodes(TreeNode *root)
  {
    return count(root, 0);
  }
};

int Solution::count(TreeNode *root, int c = 0)
{
  if (!root)
    return 0;
  if (root)
    c++;

  c += count(root->left);
  c += count(root->right);

  return c;
}

File: python/leetcode/222. Count Complete Tree Nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def countNodes(self, root: TreeNode) -> int:
        return self.count(root, 0)

    def count(self, root, c):
        if root is None:
            return 0
        if root:
            c += 1
        c += self.count(root.left, 0)
        c += self.count(root.right, 0)

        return c